### DISCUSSION TITLE
HTML ContentType with charset?

### DISCUSSION THREAD

**Discussion by troelsarvin** [tokio-rs/axum#942] [Q&A] [answered]
I can construct an HTTP header with the content type like this:
```
TypedHeader(ContentType::html())
```
That results in HTTP header "Content-Type: text/html".
What do I do if I need a "Content-Type: text/html; charset=utf-8" header?

**Comment by davidpdrsn** [comment, marked as answer]
You can always fallback to creating a `HeaderMap` manually. See https://docs.rs/axum/latest/axum/response/index.html#building-responses

**Comment by jplatte** [comment]
There doesn't seem to be a way to set the charset through the `ContentType` typed header. I'm sure they would accept a PR for this though. Repo [here](https://github.com/hyperium/headers).

**Reply by troelsarvin** [reply]
Good idea; for now, I've at least created a ticket about it: https://github.com/hyperium/headers/issues/108
